dev/devicelab/bin/tasks/gradle_plugin_test.dart

```diff
@@ -12,88 +12,125 @@ import 'package:flutter_devicelab/framework/utils.dart';
-          return _failure('Gradle output should contain a readable error message', result);
-      }
-
-      section('flutter build apk on build script with error');
-      {
-        final ProcessResult result = await project.resultOfFlutterCommand('build', <String>['apk']);
-        if (result.exitCode == 0)
-          return _failure('flutter build apk should fail when Gradle does', result);
-        final String output = result.stdout + '\n' + result.stderr;
-        if (!output.contains('Build failed') || !output.contains('builTypes'))
-          return _failure('flutter build apk output should contain a readable Gradle error message', result);
-        if (_hasMultipleOccurrences(output, 'builTypes'))
-          return _failure('flutter build apk should not invoke Gradle repeatedly on error', result);
-      }
-
-      section('gradlew assembleDebug on plugin example');
-      await pluginProject.runGradleTask('assembleDebug');
-      if (!pluginProject.hasDebugApk)
-        return new TaskResult.failure('Gradle did not produce an apk file at the expected place');
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleDebug');
+        await project.runGradleTask('assembleDebug');
+        errorMessage = _validateSnapshotDependency(project, 'build/app.dill');
+        if (errorMessage != null) {
+          throw new TaskResult.failure(errorMessage);
+        }
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleDebug no-preview-dart-2');
+        await project.runGradleTask('assembleDebug',
+            options: <String>['-Ppreview-dart-2=false']);
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleProfile');
+        await project.runGradleTask('assembleProfile');
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleRelease');
+        await project.runGradleTask('assembleRelease');
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleLocal (custom debug build)');
+        await project.addCustomBuildType('local', initWith: 'debug');
+        await project.runGradleTask('assembleLocal');
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleBeta (custom release build)');
+        await project.addCustomBuildType('beta', initWith: 'release');
+        await project.runGradleTask('assembleBeta');
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew assembleFreeDebug (product flavor)');
+        await project.addProductFlavor('free');
+        await project.runGradleTask('assembleFreeDebug');
+
+        await project.introduceError();
+      });
+
+      await runTest((project, pluginProject) async {
+        section('gradlew on build script with error');
+        {
```

Comment:
This extra scoping is not needed anymore.

Reply:
done.